src/_pytest/compat.py

```diff
@@ -45,9 +45,7 @@ def iscoroutinefunction(func):
     Note: copied and modified from Python 3.5's builtin couroutines.py to avoid import asyncio directly,
     which in turns also initializes the "logging" module as side-effect (see issue #8).
     """
-    return getattr(func, "_is_coroutine", False) or (
-        hasattr(inspect, "iscoroutinefunction") and inspect.iscoroutinefunction(func)
-    )
+    return inspect.iscoroutinefunction(func) or getattr(func, "_is_coroutine", False)
```

Comment:
BTW, I think if it would be nice to clarify the docstring of this function a bit, if you have time. I would write:

```
Return True if func is a coroutine function (a function defined with async def syntax, and doesn't contain yield),
or a function decorated with @asyncio.coroutine.

Note: copied and modified from Python 3.5's builtin couroutines.py to avoid import asyncio directly,
which in turns also initializes the "logging" module as side-effect (see issue #8).
```